enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,338 @@
+- [Platform Operators Proposal](https://github.com/openshift/enhancements/pull/1170)
+- [Capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+---
+
+# Introduce new platform type "External" in the OpenShift specific Infrastructure resource
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This EP describes an addition of the new special platform type, referred here as "External".
+
+Such new platform type would serve as a signal when OpenShift is running on an infrastructure platform
+that will receive supplemental components from the infrastructure provider.
+
+“External” is different from our current platform types in that it will signal that the infrastructure
+is specifically not “None” or any of the known providers (eg AWS, GCP, etc).
+
+## Motivation
+
+At the moment [Infrastructure](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L14)
+resource serves as a main source of information about the underlying infrastructure provider and provider specific parameters.
+Specifically [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resources holds that data.
+
+Given that PlatformSpec and PlatformStatus defined as a "discriminated unions" and require to have platform type encoded within
+"openshift/api" beforehand, it creates significant footprint and plenty of effort from RedHat engineers to initial technical enablement of
+a new cloud-provider and mostly impossible without RedHat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
+requires an information about, at least, presence of underlying cloud-provider - "None" platform does not suite well as a signal in such situation.
+
+Having special, built-in and somewhat generic, platform-type which will signal about presence of underlying infrastructure platform without
+platform specific details might help to reduce a number of pull-requests across OCP repositories
+and simplify initial integration work for external to RedHat contributors.
```

Comment:
minor nit
```suggestion
and simplify the initial integration work for non-Red Hat contributors.
```

additionally, i think we should also mention that the "external" platform will help us to identify support cases, and alsp where the responsibility lies with respect to platform integrators that utilize the new platform type.